### DISCUSSION TITLE
Ceres Releases

### DISCUSSION THREAD

**Discussion by ghost** [ceres-solver/ceres-solver#742] [General]
Hey there :)

I was wondering about the release cycle of ceres. The latest tag is 2.0.0, from October 2020. I needed [this fix](https://ceres-solver-review.googlesource.com/c/ceres-solver/+/18500), so I build ceres from master. I am not sure if there is a change log for consumers of the library, or if the latest master could could contain broken/wip parts. So I am afraid that things might break by switching from 2.0.0. Also the documentation is tagged for 2.0 if I am not mistaken. Would it be possible to create more frequent intermediate releases? Or are there any reasons not to?

I greatly appreciate that ceres is open source, it is a marvelous library!

**Comment by alexsmac** [comment]
Thanks @TobiFisch, we try hard to ensure that *every* commit on master could be a candidate release, so there should not be a 'bad' or broken commit.  We also try to ensure that we bump the major/minor version numbers when we change the API in a way that would not be backwards compatible.  In short, you should be able to roll forward to any commit in master with the same Ceres major/minor version without issue.

We are about to merge a CL introducing Manifolds (which will replace LocalParameterizations) and which bumps the Ceres version to 2.1.  However, it doesn't break backwards compatibility (i.e. it supports both Manifolds and LocalParameterizations).  It does however deprecate the LocalParameterization APIs, and we will remove support for them in future, likely as part of the next release.  We will have a discussion amongst the maintainers about when exactly we cut the next release, but you are right that it has been a pretty long time since 2.0.0.

**Comment by sandwichmaker** [comment]
@TobiFisch you can expect a new alpha in early Feb. In the interim you should be able to use the version at HEAD without any problems.

**Comment by leiwanporsche** [comment]
Hello, Could I ask a question here? Is that able to install multi versions of ceres solver under Ubuntu 20.04? Since I found the newest version of ceres solver changes many functions. But I want to use Cuda to accelerate part of the program. So I hope to install the 2.1.0 and the newest version on the main (maybe release as 2.2.0).
